packages/loader/lib/index.js

```diff
@@ -37,15 +48,44 @@ export function loader(value, callback) {
     )
   }
 
-  let process = cache.get(config)
+  let map = cache.get(compiler)
+
+  if (!map) {
+    map = new Map()
+    cache.set(compiler, map)
+  }
+
+  let process = map.get(hash)
 
   if (!process) {
     process = createFormatAwareProcessors(config).process
-    cache.set(config, process)
+    map.set(hash, process)
   }
 
   process({value, path: this.resourcePath}).then((file) => {
     callback(null, file.value, file.map)
     return file
   }, callback)
 }
+
+/**
+ * @param {Options} options
+ */
+function getOptionsHash(options) {
+  const hash = createHash('sha256')
```

Comment:
Here speed would be of value, it may be worth considering a blake2 hash

```suggestion
  const hash = createHash('blake2s256')
```

Reply:
I’m a bit uneasy with a digest format that is [not frequently used](https://github.com/search?l=JavaScript&q=%22blake2s256%22&type=Code).
Searching Google or markdown on GitHub seems to almost exclusively include it in lists of all possible algorithms, not in code like us here or in tutorials or so.

I was also uneasy about a bundled/external dependency. It does seem like `blake2s256` is supported in all maintained versions of Node.js though, as it seems to have landed in OpenSSL 1.1.1. So it seems okay, but I know that there are smaller Node.js builds out there as well?

I’m also assuming that the time factor is fine when using sha256, options objects should be rather small (especially as functions omitted) and performance bottlenecks of webpack and MDX are in other areas.

I do see room for improvement around hashing the options. In the area of preventing crashes on cyclical references in options, adding support for functions, and using a package to hash an options object which likely already exists, 

Given that you’re :+1: on this PR without this change, I’ll merge it as is, but I’m open to discussing improvements to this PRs functionality.

Reply:
the 2b variant is also an option, and is more common https://github.com/search?q=%22blake2b%22&type=Code
The reason I suggested 2s is it matches the hash size, and it optimized for 16/32 bit processors.
While 2b has a larger hash size and is optimized for 64 bit.